frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BaseBucketPickerPopover.tsx

```diff
@@ -0,0 +1,90 @@
+import { useMemo } from "react";
+import PopoverWithTrigger from "metabase/components/PopoverWithTrigger/TippyPopoverWithTrigger";
+import * as Lib from "metabase-lib";
+import {
+  TriggerButton,
+  TriggerIcon,
+  SelectList,
+  SelectListItem,
+} from "./BaseBucketPickerPopover.styled";
+
+type NoBucket = null;
+
+export type BucketListItem = Lib.BucketDisplayInfo & {
+  bucket: Lib.Bucket | NoBucket;
+};
+
+export interface BaseBucketPickerPopoverProps {
+  query: Lib.Query;
+  stageIndex: number;
+  items: BucketListItem[];
+  selectedBucket: Lib.Bucket | NoBucket;
+  isEditing: boolean;
+  triggerLabel?: string;
+  checkBucketIsSelected: (item: BucketListItem) => boolean;
+  renderTriggerContent: (bucket?: Lib.BucketDisplayInfo) => void;
+  onSelect: (column: Lib.Bucket | NoBucket) => void;
+}
+
+export function BaseBucketPickerPopover({
+  query,
+  stageIndex,
+  items,
+  selectedBucket,
+  isEditing,
+  triggerLabel,
+  checkBucketIsSelected,
+  renderTriggerContent,
+  onSelect,
+}: BaseBucketPickerPopoverProps) {
+  const defaultBucket = useMemo(
+    () => items.find(item => item.default)?.bucket,
+    [items],
+  );
+
+  const triggerContentBucket = isEditing ? selectedBucket : defaultBucket;
+  const triggerContentBucketDisplayInfo = triggerContentBucket
+    ? Lib.displayInfo(query, stageIndex, triggerContentBucket)
+    : undefined;
+
+  return (
+    <PopoverWithTrigger
+      renderTrigger={({ onClick }) => (
+        <TriggerButton
+          aria-label={triggerLabel}
+          onClick={onClick}
+          // Compat with E2E tests around MLv1-based components
+          // Prefer using a11y role selectors
+          data-testid="dimension-list-item-binning"
```

Comment:
Should be removable once QB `SummarizeSidebar` is ported to MLv2 as well (in progress)